internal/operator-controller/rukpak/bundle/config.go

```diff
@@ -6,119 +30,317 @@ import (
+			}
+			if str == installNamespace {
+				return fmt.Errorf("invalid value %q: watchNamespace must be different from %q (the install namespace) because this operator uses SingleNamespace install mode to watch a different namespace", str, installNamespace)
+			}
+			return nil
+		},
+	})
 
-	// if watchNamespace is set then ensure it is a valid namespace
-	if config.WatchNamespace != nil {
-		if errs := validation.IsDNS1123Subdomain(*config.WatchNamespace); len(errs) > 0 {
-			return fmt.Errorf("invalid 'watchNamespace' %q: namespace must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character", *config.WatchNamespace)
-		}
+	if err := compiler.AddResource(configSchemaID, schema); err != nil {
+		return fmt.Errorf("failed to load schema: %w", err)
 	}
 
-	// only accept install namespace if OwnNamespace install mode is supported
-	if config.WatchNamespace != nil && *config.WatchNamespace == installNamespace &&
-		!bundleInstallModeSet.Has(v1alpha1.InstallMode{Type: v1alpha1.InstallModeTypeOwnNamespace, Supported: true}) {
-		return fmt.Errorf("invalid 'watchNamespace' %q: must not be install namespace (%s)", *config.WatchNamespace, installNamespace)
+	compiledSchema, err := compiler.Compile(configSchemaID)
+	if err != nil {
+		return fmt.Errorf("failed to compile schema: %w", err)
 	}
 
-	// only accept non-install namespace is SingleNamespace is supported
-	if config.WatchNamespace != nil && *config.WatchNamespace != installNamespace &&
-		!bundleInstallModeSet.Has(v1alpha1.InstallMode{Type: v1alpha1.InstallModeTypeSingleNamespace, Supported: true}) {
-		return fmt.Errorf("invalid 'watchNamespace' %q: must be install namespace (%s)", *config.WatchNamespace, installNamespace)
+	if err := compiledSchema.Validate(configData); err != nil {
+		return formatSchemaError(err)
 	}
 
 	return nil
 }
 
-// isWatchNamespaceConfigSupported returns true when the bundle exposes a watchNamespace configuration. This happens when:
-// - SingleNamespace and/or OwnNamespace install modes are supported
-func isWatchNamespaceConfigSupported(bundleInstallModeSet sets.Set[v1alpha1.InstallMode]) bool {
-	return bundleInstallModeSet.HasAny(
-		v1alpha1.InstallMode{Type: v1alpha1.InstallModeTypeSingleNamespace, Supported: true},
-		v1alpha1.InstallMode{Type: v1alpha1.InstallModeTypeOwnNamespace, Supported: true},
-	)
-}
+// formatSchemaError converts technical JSON schema errors into messages users can understand.
+//
+// The JSON schema library gives us error objects, but they don't have all the details
+// we need to make helpful messages (like which field name or what value was wrong).
+// So we parse the error strings to extract that information.
+//
+// The Test_ErrorFormatting_* tests make sure this parsing works. If we upgrade the
+// JSON schema library and those tests start failing, we'll know we need to update
+// this parsing logic.
+func formatSchemaError(err error) error {
```

Comment:
`formatSchemaError` is unprecise because it converts `ValueError` into another error instance. Also, why a user could not understand the original ValueError message?

Reply:
The minimal option here would be like

```
// formatSchemaError removes technical schema file paths from validation errors
// while preserving the library's clear error messages.
func formatSchemaError(err error) error {
	msg := err.Error()
	// Strip the "jsonschema validation failed with 'file:///.../schema.json#'" prefix
	// Keep everything after the newline which has the actual error details
	if idx := strings.Index(msg, "\n"); idx != -1 {
		return fmt.Errorf("invalid configuration:%s", msg[idx:])
	}
	return fmt.Errorf("invalid configuration: %s", msg)
}
```

However, I followed your idea of `DetailedOutput()` and I think we have better messages without being so fragile. 
Give a look and let me wdyt?